Modules/_datetimemodule.c

```diff
@@ -1510,11 +1510,11 @@ call_tzinfo_method(PyObject *tzinfo, const char *name, PyObject *tzinfoarg)
                 GET_TD_SECONDS(offset) == 0 &&
                 GET_TD_MICROSECONDS(offset) < 1) ||
             GET_TD_DAYS(offset) < -1 || GET_TD_DAYS(offset) >= 1) {
-            Py_DECREF(offset);
             PyErr_Format(PyExc_ValueError, "offset must be a timedelta"
                          " strictly between -timedelta(hours=24) and"
                          " timedelta(hours=24),"
                          " not %R", offset);
+            Py_DECREF(offset);
```

Comment:
Good catch. This is a bugfix, so it should be torn out as a separate PR; we should backport that fix to the bugfix branches.

Reply:
This is not a bugfix, previously `Py_DECREF(offset)` was called before calling `PyErr_Format` because the error message did not contain `offset`.
and after I added `offset` to the error message I had to call `Py_DECREF(offset)` after so that the variable would be available.

Reply:
Ah, thanks, I missed that!